test/InSAR/InSARTest.jl

```diff
@@ -0,0 +1,180 @@
+
+    # Assert
+    test_ok = all([check_real, check_non_zero, check_monotonic]);
+
+    if !test_ok
+        println("Debug info: ", string(StackTraces.stacktrace()[1].func))
+        println("eltype(_doppler_fm_rate(x, param, x0)): ", output_type)
+        println("Is monotonically increasing: ", all(check_monotonically_increasing))
+        println("Is monotonically decreasing: ", all(check_monotonically_decreasing))
+    end
+
+    return test_ok
+end
+
+function _doppler_fm_rate_test()
+    # Load
+    image = load_test_slc_image();
+    burst_number = SARProcessing.get_burst_numbers(image)[1]
+    metadata = image.metadata;
+    tau_0 = metadata.image.slant_range_time_seconds; 
+    Delta_tau_s = 1/metadata.product.range_sampling_rate;
+    columns = 1:metadata.image.number_of_samples;
+    range_time = tau_0 .+ (collect(columns) .- 1) .* Delta_tau_s
+    fm_meta = metadata.bursts[burst_number].azimuth_fm_rate;
+
+    # Test
+    polynomial_test(SARProcessing._doppler_fm_rate, range_time, fm_meta.polynomial, fm_meta.t0)
+end
+
+function _doppler_centroid_frequency_test()
+    # Load
+    image = load_test_slc_image();
+    burst_number = SARProcessing.get_burst_numbers(image)[1]
+    metadata = image.metadata;
+    tau_0 = metadata.image.slant_range_time_seconds; 
+    Delta_tau_s = 1/metadata.product.range_sampling_rate;
+    columns = 1:metadata.image.number_of_samples;
+    range_time = tau_0 .+ (collect(columns) .- 1) .* Delta_tau_s
+    dc_meta = metadata.bursts[burst_number].doppler_centroid;
+
+    # Test
+    polynomial_test(SARProcessing._doppler_centroid_frequency, range_time, dc_meta.polynomial, dc_meta.t0)
+end
+
+function phase_ramp_test()
+    ## Arrange
+    # load image
+    image = load_test_slc_image();
+    meta_data = image.metadata;
+    burst_number = SARProcessing.get_burst_numbers(image)[1]
+
+    # get range times from columns/samples
+    orbit_states = SARProcessing.load_precise_orbit_sentinel1(PRECISE_ORBIT_TEST_FILE2)
+
+    # ## Get mid burst time and speed
+    interpolator = SARProcessing.orbit_state_interpolator(orbit_states, image)
+    mid_burst_state = SARProcessing.get_burst_mid_states(image, interpolator)
+    mid_burst_speed = SARProcessing.get_speed.(mid_burst_state)[1]
+
+    # Create some matrices of row and colomn indeces
+    image_window = SARProcessing.get_window(image)
+    columns = collect(image_window[2][1]:image_window[2][2])
+    rows = collect(image_window[1][1]:image_window[1][2])
+
+    ## Act
+    ramp = SARProcessing.phase_ramp(rows, columns, burst_number, mid_burst_speed, meta_data);
+
+    ## Assert
+    # check if output is real
+    output_type = eltype(ramp);
+    check_real = output_type <: Real
+
+    # check that the ramp is not all zeros
+    check_non_zero = sum(ramp .!= 0) != 0;
+
+    # check that length of each element is 1 for reramp * deramp
+    check_ramp_cancellation_length = all(length.(exp.(-ramp .* im) .* exp.(ramp .* im)) .== 1)
```

Comment:
`exp.(-x .* im) .* exp.(x .* im)` is one of all x. Test is true no matter the ramp